### PR TITLE
Correctly merging "require" section when having branch alias

### PR DESCRIPTION
**Pull request by astax-t**
Fixes https://github.com/wikimedia/composer-merge-plugin/issues/62

### CODE REVIEW

File: src/Merge/ExtraPackage.php [hunk 1]
```diff
@@ -125,13 +127,20 @@ protected function loadPackage($json)
...
-     * @param RootPackage $root
+     * @param RootPackageInterface $top
      * @param PluginState $state
      */
-    public function mergeInto(RootPackage $root, PluginState $state)
+    public function mergeInto(RootPackageInterface $top, PluginState $state)
     {
-        $this->mergeRequires($root, $state);
-        $this->mergeDevRequires($root, $state);
+        if ($top instanceof RootAliasPackage)
```
Comment (bd808) [inline src/Merge/ExtraPackage.php]: PluginState::getRootPackage() already does this unwrap. Did I mess something up in the refactoring patch that dropped usage of it in a key location?

Comment (astax-t) [inline src/Merge/ExtraPackage.php]: Yes, it does and that's the problem - it unwraps alias too early. Note the call to `mergeRequires` below needs $top, which wouldn't be available if alias is unwrapped in PluginState.

But this code can be replaced by `$root = $state->getRootPackage();`. I wasn't sure what's the best way. I just think that the current code is easier to understand as it shows clear connection between $top and $root.

**Comment by bd808** [comment]
I used this as a starting point for #67 which tries to go a bit deeper into the problem.

**Comment by astax-t** [comment]
Thanks

**Pull request closed without merging**